### PR TITLE
Support for Batch Read API

### PR DESCRIPTION
**Pull request by snehashah16**

### CODE REVIEW

**Comment by vkedia** [comment]
cc @garrettjonesgoogle @andreamlin PTAL

**Comment by vkedia** [comment]
Note that we also need to update the proto version before this can be merged.

File: google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerImpl.java [hunk 1]
```diff
@@ -1086,13 +1114,24 @@ private ResultSet readInternal(
         KeySet keys,
         Iterable<String> columns,
         ReadOption... options) {
+      Options readOptions = Options.fromReadOptions(options);
+      return readInternalWithOptions(
+          table, index, keys, columns, readOptions, null /*partitionToken*/);
+    }
+
```
[the remark this answers is not in the document]
Comment (snehashah16) [reply google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerImpl.java]: This method is building the request, making RPC and processing the result. I will address this issue in a separate PR (mostly after the release) with adequate testing.

File: google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerImpl.java [hunk 2]
```diff
@@ -987,6 +1003,16 @@ private ResultSet executeQueryInternal(
         Statement statement,
         com.google.spanner.v1.ExecuteSqlRequest.QueryMode queryMode,
         QueryOption... options) {
+      Options readOptions = Options.fromQueryOptions(options);
+      return executeQueryInternalWithOptions(
+          statement, queryMode, readOptions, null /*partitionToken*/);
+    }
+
```
[the remark this answers is not in the document]
Comment (snehashah16) [reply google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerImpl.java]: This method is building the request, making RPC and processing the result. I will address this issue in a separate PR (mostly after the release) with adequate testing.

**Comment by snehashah16** [comment]
Hey garrett, will we need an update to pom.xml (for latest grpc & proto versions) after the new release process (w/ version.txt ) is done ?

**Comment by garrettjonesgoogle** [comment]
@andreamlin is taking care of this.

**Pull request merged**